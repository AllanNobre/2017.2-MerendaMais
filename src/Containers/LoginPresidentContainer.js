import { connect } from 'react-redux';
import { LoginPresidente } from '../screens';
import { modifyCPF, modifyPassword, asyncLoginCounselor } from '../actions/counselorActions';

const mapStatetoProps = state => (
  {
    cpf: state.counselor.cpf,
    password: state.counselor.password,
    message_erro: state.counselor.message_erro,
    isLoading: state.counselor.isLoading,
  }
);

export default connect(mapStatetoProps,
  { modifyCPF, modifyPassword, asyncLoginCounselor },
)(LoginPresidente);
