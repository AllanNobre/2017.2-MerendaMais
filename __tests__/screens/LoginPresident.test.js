import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// imported as a connected component!
import LoginPresidentContainer from '../../src/Containers/LoginPresidentContainer';


Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  application: {
    isLoading: false,
    message_erro: '',
  },
};

const store = mockStore(initialState);

describe('Testing LoginPresidente', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <LoginPresidentContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
