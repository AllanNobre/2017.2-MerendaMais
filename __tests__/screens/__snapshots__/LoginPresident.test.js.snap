// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginPresidente renders as expected 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Header
    backButton={false}
    subTitle=""
    title="Merenda +"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 6,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "gray",
          "borderRadius": 7,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingLeft": 2,
          "paddingRight": 4,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="black"
        name="email"
        size={32}
        style={
          Object {
            "height": 30,
            "margin": 5,
            "width": 30,
          }
        }
      />
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Email"
        returnKeyType="next"
        underlineColorAndroid="transparent"
        width={280}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#FAFAFA",
          "borderColor": "gray",
          "borderRadius": 7,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 30,
          "paddingLeft": 2,
          "paddingRight": 4,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="black"
        name="lock"
        size={32}
        style={
          Object {
            "height": 30,
            "margin": 5,
            "width": 30,
          }
        }
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Senha"
        returnKeyType="go"
        secureTextEntry={true}
        underlineColorAndroid="transparent"
        width={280}
      />
    </View>
    <TouchableOpacity
      activeOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#FF9500",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginBottom": 0,
          "marginTop": 50,
          "paddingHorizontal": 133,
          "paddingVertical": 18,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Entrar
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.6}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        É um conselheiro?
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "blue",
            }
          }
        >
          Clique aqui
        </Text>
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#a9a9a9",
        "borderTopWidth": 1,
        "flex": 0.7,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.6}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Ainda não se cadastrou?
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "blue",
            }
          }
        >
           Cadastrar-se
        </Text>
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
